File: ExampleOBJC/OBJCExample/ViewController.m
//
//  ViewController.m
//  OBJCExample
//
//  Created by Aurelien on 28/05/2019.
//  Copyright © 2019 chronotruck. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <FPNTextFieldDelegate>

@property (strong, nonatomic) FPNTextField *phoneNumberTextField;

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view.

		//		NSBundle.FlagIcons = [NSBundle bundleForClass: [self class]];
	_phoneNumberTextField = [[FPNTextField alloc] initWithFrame: CGRectMake(0, 0, self.view.bounds.size.width - 16, 50)];
	_phoneNumberTextField.borderStyle = UITextBorderStyleRoundedRect;
	_phoneNumberTextField.displayMode = 0;
	_phoneNumberTextField.delegate = self;
		//	_phoneNumberTextField.flagSize = CGSizeMake(35, 35);
		//	_phoneNumberTextField.flagButtonEdgeInsets = UIEdgeInsetsMake(10, 10, 10, 10);
	_phoneNumberTextField.hasPhoneNumberExample = TRUE;
		//	_phoneNumberTextField.placeholder = @"Phone Number";
		//	[_phoneNumberTextField setCountriesIncluding: @[@(FPNOBJCCountryKeyAM),@(FPNOBJCCountryKeyBW),@(FPNOBJCCountryKeyBA)]];
		//	[_phoneNumberTextField setCountriesWithExcluding: @[@(FPNOBJCCountryKeyAM),@(FPNOBJCCountryKeyBW),@(FPNOBJCCountryKeyBA)]];
		//	[_phoneNumberTextField setFlagFor: FPNOBJCCountryKeyFR];
	[_phoneNumberTextField setWithPhoneNumber: @"+33612345678"];

	[self.view addSubview: _phoneNumberTextField];

	_phoneNumberTextField.center = self.view.center;
}

- (void)fpnDidSelectCountryWithName:(NSString * _Nonnull)name dialCode:(NSString * _Nonnull)dialCode code:(NSString * _Nonnull)code {
	NSLog(@"%@ %@ %@", name, dialCode, code);
}

- (void)fpnDidValidatePhoneNumberWithTextField:(FPNTextField * _Nonnull)textField isValid:(BOOL)isValid {
	UIImage *img = isValid ? [UIImage imageNamed: @"success"] : [UIImage imageNamed: @"error"];

	_phoneNumberTextField.rightViewMode = UITextFieldViewModeAlways;
	_phoneNumberTextField.rightView = [[UIImageView alloc] initWithImage: img];

	NSLog(@"is valid: %@", isValid ? @"Yes" : @"No");
	NSLog(@"E164 Format: %@", [_phoneNumberTextField getFormattedPhoneNumberWithFormat: FPNFormatE164]);
	NSLog(@"International Format: %@", [_phoneNumberTextField getFormattedPhoneNumberWithFormat: FPNFormatInternational]);
	NSLog(@"National Format: %@", [_phoneNumberTextField getFormattedPhoneNumberWithFormat: FPNFormatNational]);
	NSLog(@"RFC3966 Format: %@", [_phoneNumberTextField getFormattedPhoneNumberWithFormat: FPNFormatRFC3966]);
	NSLog(@"Raw: %@", [_phoneNumberTextField getRawPhoneNumber]);
}

- (void)fpnDisplayCountryList {

}

@end

File: CarthageExample/Example/ViewController.swift
//
//  ViewController.swift
//  Example
//
//  Created by Aurélien Grifasi on 06/08/2017.
//  Copyright (c) 2017 Aurélien Grifasi. All rights reserved.
//

import UIKit
import FlagPhoneNumber

class ViewController: UIViewController {

	var phoneNumberTextField: FPNTextField!

	override func viewDidLoad() {
		super.viewDidLoad()

		title = "FlagPhoneNumber"

		// To use your own flag icons, uncommment the line :
		//		Bundle.FlagIcons = Bundle(for: ViewController.self)

		phoneNumberTextField = FPNTextField(frame: CGRect(x: 0, y: 0, width: view.bounds.width - 16, height: 50))
		phoneNumberTextField.borderStyle = .roundedRect

		// Comment this line to not have access to the country list
		phoneNumberTextField.parentViewController = self
		phoneNumberTextField.delegate = self

		// Custom the size/edgeInsets of the flag button
		phoneNumberTextField.flagSize = CGSize(width: 35, height: 35)
		phoneNumberTextField.flagButtonEdgeInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)

		// Example of customizing the textField input accessory view
		//		let items = [
		//			UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.save, target: self, action: nil),
		//			UIBarButtonItem(title: "Item 1", style: .plain, target: self, action: nil),
		//			UIBarButtonItem(title: "Item 2", style: .plain, target: self, action: nil)
		//		]
		//		phoneNumberTextField.textFieldInputAccessoryView = getCustomTextFieldInputAccessoryView(with: items)

		// The placeholder is an example phone number of the selected country by default. You can add your own placeholder :
		phoneNumberTextField.hasPhoneNumberExample = true
		//		phoneNumberTextField.placeholder = "Phone Number"

		// Set the country list
		//		phoneNumberTextField.setCountries(including: [.FR, .ES, .IT, .BE, .LU, .DE])

		// Exclude countries from the list
		//		phoneNumberTextField.setCountries(excluding: [.AM, .BW, .BA])

		// Set the flag image with a region code
		//		phoneNumberTextField.setFlag(for: "FR")

		// Set the phone number directly
		//		phoneNumberTextField.set(phoneNumber: "+33612345678")

		view.addSubview(phoneNumberTextField)

		phoneNumberTextField.center = view.center
	}

	private func getCustomTextFieldInputAccessoryView(with items: [UIBarButtonItem]) -> UIToolbar {
		let toolbar: UIToolbar = UIToolbar()

		toolbar.barStyle = UIBarStyle.default
		toolbar.items = items
		toolbar.sizeToFit()

		return toolbar
	}
}

extension ViewController: FPNTextFieldDelegate {

	func fpnDidValidatePhoneNumber(textField: FPNTextField, isValid: Bool) {
		textField.rightViewMode = .always
		textField.rightView = UIImageView(image: isValid ? #imageLiteral(resourceName: "success") : #imageLiteral(resourceName: "error"))

		print(
			isValid,
			textField.getFormattedPhoneNumber(format: .E164) ?? "E164: nil",
			textField.getFormattedPhoneNumber(format: .International) ?? "International: nil",
			textField.getFormattedPhoneNumber(format: .National) ?? "National: nil",
			textField.getFormattedPhoneNumber(format: .RFC3966) ?? "RFC3966: nil",
			textField.getRawPhoneNumber() ?? "Raw: nil"
		)
	}

	func fpnDidSelectCountry(name: String, dialCode: String, code: String) {
		print(name, dialCode, code)
	}
}
